File: resources/expected/ros2/isolette_inverted_topics/lax/src/isolette_cpp_pkg/src/base_code/thermostat_regulate_temperature_manage_regulator_mode_mrmt_base_src.cpp
#include "isolette_cpp_pkg/base_headers/thermostat_regulate_temperature_manage_regulator_mode_mrmt_base_src.hpp"

//========================================================
// Re-running Codegen will overwrite changes to this file
//========================================================

thermostat_regulate_temperature_manage_regulator_mode_mrmt_base::thermostat_regulate_temperature_manage_regulator_mode_mrmt_base() : Node("thermostat_regulate_temperature_manage_regulator_mode_mrmt")
{
    cb_group_ = this->create_callback_group(rclcpp::CallbackGroupType::Reentrant);
    subscription_options_.callback_group = cb_group_;

    // Setting up connections
    thermostat_regulate_temperature_manage_regulator_mode_mrmt_current_tempWstatus_subscription_ = this->create_subscription<isolette_cpp_pkg_interfaces::msg::TempWstatusimpl>(
        "temperature_sensor_cpi_thermostat_current_tempWstatus",
        1,
        std::bind(&thermostat_regulate_temperature_manage_regulator_mode_mrmt_base::handle_current_tempWstatus, this, std::placeholders::_1), subscription_options_);

    thermostat_regulate_temperature_manage_regulator_mode_mrmt_interface_failure_subscription_ = this->create_subscription<isolette_cpp_pkg_interfaces::msg::FailureFlagimpl>(
        "thermostat_regulate_temperature_manage_regulator_interface_mrit_interface_failure",
        1,
        std::bind(&thermostat_regulate_temperature_manage_regulator_mode_mrmt_base::handle_interface_failure, this, std::placeholders::_1), subscription_options_);

    thermostat_regulate_temperature_manage_regulator_mode_mrmt_internal_failure_subscription_ = this->create_subscription<isolette_cpp_pkg_interfaces::msg::FailureFlagimpl>(
        "thermostat_regulate_temperature_manage_regulator_mode_mrmt_internal_failure",
        1,
        std::bind(&thermostat_regulate_temperature_manage_regulator_mode_mrmt_base::handle_internal_failure, this, std::placeholders::_1), subscription_options_);

    thermostat_regulate_temperature_manage_regulator_mode_mrmt_regulator_mode_publisher_ = this->create_publisher<isolette_cpp_pkg_interfaces::msg::RegulatorMode>(
        "thermostat_regulate_temperature_manage_regulator_mode_mrmt_regulator_mode",
        1);

    // timeTriggered callback timer
    periodTimer_ = this->create_wall_timer(std::chrono::milliseconds(1000),
        std::bind(&thermostat_regulate_temperature_manage_regulator_mode_mrmt_base::timeTriggered, this), cb_group_);

}

//=================================================
//  C o m m u n i c a t i o n
//=================================================

void thermostat_regulate_temperature_manage_regulator_mode_mrmt_base::handle_current_tempWstatus(const isolette_cpp_pkg_interfaces::msg::TempWstatusimpl::SharedPtr msg)
{
    current_tempWstatus_msg_holder = msg;
}

void thermostat_regulate_temperature_manage_regulator_mode_mrmt_base::handle_interface_failure(const isolette_cpp_pkg_interfaces::msg::FailureFlagimpl::SharedPtr msg)
{
    interface_failure_msg_holder = msg;
}

void thermostat_regulate_temperature_manage_regulator_mode_mrmt_base::handle_internal_failure(const isolette_cpp_pkg_interfaces::msg::FailureFlagimpl::SharedPtr msg)
{
    internal_failure_msg_holder = msg;
}

isolette_cpp_pkg_interfaces::msg::TempWstatusimpl::SharedPtr thermostat_regulate_temperature_manage_regulator_mode_mrmt_base::get_current_tempWstatus() {
    return current_tempWstatus_msg_holder;
}

isolette_cpp_pkg_interfaces::msg::FailureFlagimpl::SharedPtr thermostat_regulate_temperature_manage_regulator_mode_mrmt_base::get_interface_failure() {
    return interface_failure_msg_holder;
}

isolette_cpp_pkg_interfaces::msg::FailureFlagimpl::SharedPtr thermostat_regulate_temperature_manage_regulator_mode_mrmt_base::get_internal_failure() {
    return internal_failure_msg_holder;
}

void thermostat_regulate_temperature_manage_regulator_mode_mrmt_base::put_regulator_mode(isolette_cpp_pkg_interfaces::msg::RegulatorMode msg)
{
    thermostat_regulate_temperature_manage_regulator_mode_mrmt_regulator_mode_publisher_->publish(msg);
}

File: resources/expected/ros2/python-isolette/lax/src/isolette_py_pkg/isolette_py_pkg/base_code/thermostat_regulate_temperature_manage_regulator_mode_mrmt_base_src.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from queue import Queue
from isolette_py_pkg.user_code.thermostat_regulate_temperature_manage_regulator_mode_mrmt_src import *
from rclpy.callback_groups import ReentrantCallbackGroup
from isolette_py_pkg_interfaces.msg import TempWstatusimpl
from isolette_py_pkg_interfaces.msg import FailureFlagimpl
from isolette_py_pkg_interfaces.msg import RegulatorMode

#========================================================
# Re-running Codegen will overwrite changes to this file
#========================================================

class thermostat_regulate_temperature_manage_regulator_mode_mrmt_base(Node):
    def __init__(self):
        super().__init__("thermostat_regulate_temperature_manage_regulator_mode_mrmt")

        self.cb_group_ = ReentrantCallbackGroup()

        # Setting up connections
        self.thermostat_regulate_temperature_manage_regulator_mode_mrmt_current_tempWstatus_subscription_ = self.create_subscription(
            TempWstatusimpl,
            "thermostat_regulate_temperature_manage_regulator_mode_mrmt_current_tempWstatus",
            self.handle_current_tempWstatus,
            1,
            callback_group=self.cb_group_)

        self.thermostat_regulate_temperature_manage_regulator_mode_mrmt_interface_failure_subscription_ = self.create_subscription(
            FailureFlagimpl,
            "thermostat_regulate_temperature_manage_regulator_mode_mrmt_interface_failure",
            self.handle_interface_failure,
            1,
            callback_group=self.cb_group_)

        self.thermostat_regulate_temperature_manage_regulator_mode_mrmt_internal_failure_subscription_ = self.create_subscription(
            FailureFlagimpl,
            "thermostat_regulate_temperature_manage_regulator_mode_mrmt_internal_failure",
            self.handle_internal_failure,
            1,
            callback_group=self.cb_group_)

        self.thermostat_regulate_temperature_manage_regulator_mode_mrmt_regulator_mode_publisher_1 = self.create_publisher(
            RegulatorMode,
            "thermostat_regulate_temperature_manage_regulator_interface_mrit_regulator_mode",
            1)

        self.thermostat_regulate_temperature_manage_regulator_mode_mrmt_regulator_mode_publisher_2 = self.create_publisher(
            RegulatorMode,
            "thermostat_regulate_temperature_manage_heat_source_mhst_regulator_mode",
            1)

        # timeTriggered callback timer
        self.periodTimer_ = self.create_timer(1000, self.timeTriggered, callback_group=self.cb_group_)

    def timeTriggered(self):
        pass

    #=================================================
    #  C o m m u n i c a t i o n
    #=================================================

    def handle_current_tempWstatus(self, msg):
        typedMsg = TempWstatusimpl()
        typedMsg.data = msg
        self.current_tempWstatus_msg_holder = typedMsg

    def handle_interface_failure(self, msg):
        typedMsg = FailureFlagimpl()
        typedMsg.data = msg
        self.interface_failure_msg_holder = typedMsg

    def handle_internal_failure(self, msg):
        typedMsg = FailureFlagimpl()
        typedMsg.data = msg
        self.internal_failure_msg_holder = typedMsg

    def get_current_tempWstatus(self):
        return self.current_tempWstatus_msg_holder

    def get_interface_failure(self):
        return self.interface_failure_msg_holder

    def get_internal_failure(self):
        return self.internal_failure_msg_holder

    def put_regulator_mode(self, msg):
        typedMsg = RegulatorMode()
        typedMsg.data = msg
        thermostat_regulate_temperature_manage_regulator_mode_mrmt_regulator_mode_publisher_1.publish(typedMsg)
        thermostat_regulate_temperature_manage_regulator_mode_mrmt_regulator_mode_publisher_2.publish(typedMsg)

